### ISSUE TITLE
typos inside cfg macros go undetected

### ISSUE THREAD

**Issue by JOE1994** [rust-lang/rust-clippy#5051]
The below program has typos inside the `cfg` macro call.
(`feeture` instead of `feature`)

The below program compiles without any errors, and `clippy` also does not print any warnings.
```rust
pub fn main() {
    if cfg!(feeture="EXT1") {
        println!("EXTRA FEATURE1 ADDED");
    }
    #[cfg(feeture="EXT2")] {
        println!("EXTRA FEATURE2 ADDED");
    }
}
```

`cfg` macro seems to be a compiler built-in macro (source code not available?). 
I found this issue while debugging my Rust code.

**Comment by mati865** [comment]
Only the "enabled" code is being checked by rustc. You should get what you expect by pressing `--all-features` argument to Cargo.

**Comment by JOE1994** [comment]
I apologize for not phrasing the issue well.. I am aware that only the enabled code is checked by rustc.
The issue I wanted to talk about was that `feeture` is an invalid configuration option, but neither rustc nor clippy complains about it.

**Comment by flip1995** [comment]
A lint with a whitelist of possible configurations would be possible. The problem is, that you can pass anything as a config through [`--cfg`](https://doc.rust-lang.org/rustc/command-line-arguments.html#--cfg-configure-the-compilation-environment), so with `rustc --cfg 'feeture="EXT1"' main.rs` the code would be totally valid.

**Comment by JOE1994** [comment]
@flip1995 I see.. Can we keep this issue open for a bit to get others' feedback also? I'm curious how others would think of this issue 😃

**Comment by vallentin** [comment]
I'd say, have Clippy produce a warning if there's a high similarity. So if `feeture` is used, then Clippy should say "Are you sure you didn't mean 'feature'?", since they're highly similar.

Same with say `target_familly` (double L) or `target_pointer_with` (missing d).

**Comment by flip1995** [comment]
We already do something similar with `unknown_clippy_lints`, so this should be quite easy to implement. We'd need a list of config options though.
